### ISSUE TITLE
Slider component (possibly TabBar as well) computing incorrect width in Chrome

### ISSUE THREAD

**Issue by jrajav** [jamesmfriedman/rmwc#36]
Apologies for the very vague issue - I am working on a reproduction. Just logging it in case you with more expertise might already know what's going on.

I have a relatively complex flex layout with multiple nested rows and columns. I've placed a Slider deep in that hierarchy and given it an explicit width of 100px. It works just fine on Firefox, and on Chrome if I resize the page after initial render - however, on Chrome without resizing the page at all, it seems to compute the client rect incorrectly for translating the thumb control:

https://i.imgur.com/36LZWv4.png

This persists even after interacting with the control, as well as attempting to force a reflow on the parent container. Have tried various display modes and position: relative on the parent container as well - nothing works.

This problem also seems to be present with the bottom border of tabs on the TabBar. The computed rect width seems to be almost the full width of the window or even more.

Trying to get a reproduction outside of my app, will update if successful.

**Comment by jamesmfriedman** [comment]
Let me know if you get some more details on it. RMWC simply wraps the existing MDC apis. Occasionally, depending on the component, there might be a bug where these 2 things are out of sync, or the problem could exist only in MDC. It makes it a bit challenging to hunt down where the bug lives, but if you can post a repro case, that would be a good start in helping identify the issue.

**Comment by jrajav** [comment]
No repro yet, but one additional bit of info: The problem is fixed by deferring initial render until after the app has had a chance to load the full CSS. It seems visually that the computed width of MDC Slider is based on the unstyled content width (full page width since everything is just display: block).

I am going to just guess that whatever reflow-detection (or forcing upon re-render) method MDC Slider uses isn't working quite right on Chrome when the CSS is applied.

**Comment by jamesmfriedman** [comment]
How are you loading the CSS?

If you’re doing something that’s loading it asynchronously then this could be the issue.

**Comment by jamesmfriedman** [comment]
A little more detail for you since this has happened to me in the past using Webpack and Create React App

- If you are in development mode, CRA uses the inline-style-loader with source maps. Javascript will load these styles, but they are effectively async, and you can't rely on them on browser load

- If you are in production mode, CRA creates a CSS file that gets loaded before the JS, and is render blocking. This means that you can be sure that styles are loaded before JS executes.

**Comment by jamesmfriedman** [comment]
@jrajav did any of that help?

**Comment by jrajav** [comment]
@jamesmfriedman Yes, and after my last reply I suspected that would be the case, as well. I hadn't made any workaround for the TabBar part of my app (had just been using a different framework for that), so I just switched that back and checked - indeed, it only works incorrectly on the dev environment, once fully compiled for prod with the packaged bundles it works fine.

Perhaps this is worth some documentation? A niche case though I guess. Closing as I suppose it's not really a problem, in any case.
